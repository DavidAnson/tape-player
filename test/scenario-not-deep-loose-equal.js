// @ts-check

"use strict";

const tape = require("tape");
require("../index.js");

tape("equal test", (test) => {
  test.equal(1, 1);
  test.equal(1, 1, "1 message");
  test.equal(1, 2);
  test.equal(1, 2, "2 message");
  test.end();
});

tape("notEqual test", (test) => {
  test.notEqual(1, 2);
  test.notEqual(1, 2, "2 message");
  test.notEqual(1, 1);
  test.notEqual(1, 1, "1 message");
  test.end();
});

tape("deepEqual test object", (test) => {
  const aa = { "property": 1 };
  const bb = { "property": 1 };
  const cc = { "property": 2 };
  test.deepEqual(aa, bb);
  test.deepEqual(aa, bb, "bb message");
  test.deepEqual(bb, cc);
  test.deepEqual(bb, cc, "cc message");
  test.end();
});

tape("notDeepEqual test object", (test) => {
  const aa = { "property": 1 };
  const bb = { "property": 1 };
  const cc = { "property": 2 };
  test.notDeepEqual(aa, bb);
  test.notDeepEqual(aa, bb, "bb message");
  test.notDeepEqual(bb, cc);
  test.notDeepEqual(bb, cc, "cb message");
  test.end();
});

tape("deepEqual test array", (test) => {
  const aa = [ 1, 2 ];
  const bb = [ 1, 2 ];
  const cc = [ 1, 2, 3 ];
  test.deepEqual(aa, bb);
  test.deepEqual(aa, bb, "bb message");
  test.deepEqual(bb, cc);
  test.deepEqual(bb, cc, "cc message");
  test.end();
});

tape("notDeepEqual test array", (test) => {
  const aa = [ 1, 2 ];
  const bb = [ 1, 2 ];
  const cc = [ 1, 2, 3 ];
  test.notDeepEqual(aa, bb);
  test.notDeepEqual(aa, bb, "bb message");
  test.notDeepEqual(bb, cc);
  test.notDeepEqual(bb, cc, "cb message");
  test.end();
});

tape("deepLooseEqual test", (test) => {
  const aa = { "property": 1 };
  const bb = { "property": "1" };
  const cc = { "property": 2 };
  test.deepLooseEqual(aa, bb);
  test.deepLooseEqual(aa, bb, "bb message");
  test.deepLooseEqual(bb, cc);
  test.deepLooseEqual(bb, cc, "cc message");
  test.end();
});

tape("notDeepLooseEqual test", (test) => {
  const aa = { "property": 1 };
  const bb = { "property": "1" };
  const cc = { "property": 2 };
  test.notDeepLooseEqual(aa, bb);
  test.notDeepLooseEqual(aa, bb, "bb message");
  test.notDeepLooseEqual(bb, cc);
  test.notDeepLooseEqual(bb, cc, "cc message");
  test.end();
});
