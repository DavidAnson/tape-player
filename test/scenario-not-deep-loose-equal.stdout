equal test
  scenario-not-deep-loose-equal.js:11:8
  Message:  should be equal
  Operator: equal
  Expected: 2
  Actual:   1
  scenario-not-deep-loose-equal.js:12:8
  Message:  2 message
  Operator: equal
  Expected: 2
  Actual:   1
notEqual test
  scenario-not-deep-loose-equal.js:19:8
  Message:  should not be equal
  Operator: notEqual
  Expected: 1
  Actual:   1
  scenario-not-deep-loose-equal.js:20:8
  Message:  1 message
  Operator: notEqual
  Expected: 1
  Actual:   1
deepEqual test
  scenario-not-deep-loose-equal.js:30:8
  Message:  should be equivalent
  Operator: deepEqual
  Expected: { property: 2 }
  Actual:   { property: 1 }
  scenario-not-deep-loose-equal.js:31:8
  Message:  cc message
  Operator: deepEqual
  Expected: { property: 2 }
  Actual:   { property: 1 }
notDeepEqual test
  scenario-not-deep-loose-equal.js:39:8
  Message:  should not be equivalent
  Operator: notDeepEqual
  Expected: { property: 1 }
  Actual:   { property: 1 }
  scenario-not-deep-loose-equal.js:40:8
  Message:  bb message
  Operator: notDeepEqual
  Expected: { property: 1 }
  Actual:   { property: 1 }
deepLooseEqual test
  scenario-not-deep-loose-equal.js:52:8
  Message:  should be equivalent
  Operator: deepLooseEqual
  Expected: { property: 2 }
  Actual:   { property: '1' }
  scenario-not-deep-loose-equal.js:53:8
  Message:  cc message
  Operator: deepLooseEqual
  Expected: { property: 2 }
  Actual:   { property: '1' }
notDeepLooseEqual test
  scenario-not-deep-loose-equal.js:61:8
  Message:  should be equivalent
  Operator: notDeepLooseEqual
  Expected: { property: '1' }
  Actual:   { property: 1 }
  scenario-not-deep-loose-equal.js:62:8
  Message:  bb message
  Operator: notDeepLooseEqual
  Expected: { property: '1' }
  Actual:   { property: 1 }

Tests:    6
Asserts:  24
Failures: 12
